### PR TITLE
Secpatch/high1 2nd attempt

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
This is an extension of #7735, which was reverted due to a prod issue. The only addition in this PR is tweaking `sidekiq` and `sidekiq-pro` versions, so that we don't upgrade to an extend that it breaks our production sidekiq workers. Because `sidekiq` and `sidekiq-pro` versions are not changed in this PR, we can be reasonably confident that it will not cause a production issue like the last, reverted PR caused.

The other gem and npm specifications have already been tested.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  no
Have you deployed to Staging? | about to
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Design Review: If applicable, have you compared the coded design to the mockups? | (N/A

### CODE REVIEW

File: services/QuillCMS/Gemfile.lock [hunk 1]
```diff
@@ -140,37 +148,37 @@ GEM
...
-      slop (~> 3.4)
+    pry (0.14.1)
+      coderay (~> 1.1)
+      method_source (~> 1.0)
     public_suffix (4.0.6)
-    puma (3.12.6)
-    pusher (1.4.3)
+    puma (4.3.8)
+      nio4r (~> 2.0)
+    pusher (2.0.1)
```
Comment (dandrabik) [inline services/QuillCMS/Gemfile.lock]: There's a major version bump of Pusher (1.4.3 -> 2.0.1). Is this necessary? Has this been tested?

https://github.com/pusher/pusher-http-ruby/blob/master/CHANGELOG.md

**Review by dandrabik — CHANGES_REQUESTED** [review]
Can we remove the Pusher update from this version bump? If not, can we confirm that this has been tested.

File: services/QuillCMS/Gemfile.lock (hunk 1 above)
Comment (happythenewsad) [reply services/QuillCMS/Gemfile.lock]: Looks like newrelic_rpm and method_source were bumped significantly as well. I'll attempt to pin them to their previous versions, but will fall back to manual testing if I can't.

**Comment by happythenewsad** [comment]
I was able to pin all three gems so that they didn't change versions.

File: services/QuillCMS/Gemfile [hunk 2]
```diff
@@ -57,7 +58,7 @@ end
 group :development, :test do
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
   gem 'factory_bot_rails', '~> 4.8.2'
-  gem 'pry'
+  gem 'pry', '~> 0.10.4'
```
Comment (dandrabik) [inline services/QuillCMS/Gemfile:61]: This is a `development` only gem, so we don't need to pin it.

**Review by dandrabik — APPROVED** [review]
Small comment about unpinning a development gem, but :+1:

File: services/QuillCMS/Gemfile (hunk 2 above)
Comment (happythenewsad) [reply services/QuillCMS/Gemfile:61]: Actually, unpinning will cause bundler to install a high version of `method-source`, and `method-source` is used by production code via the railties gem.

Comment (dandrabik) [reply services/QuillCMS/Gemfile:61]: Fine to pin then. :ship:

**Review by anathomical — APPROVED** [review]

**Pull request merged**